vignette updates involving tcplPlot and package size for cran release

cthunes opened the pull request:
- Consolidated plotting sections within Data retrieval and data retrieval via api sections into new plotting section
- Improved table, text and images with new tcplPlot parameters and visual updates
- Removed unnecessary image files from vignettes/img (though this does not affect final package size like we thought -- only the images that are stored in the vignette)
- Reduced file size of about 8 images through compression, all but one for plotting. The one was @ElizabethGilson's flag visual which was about 1.3Mb that I did compress down to 160 Kb, hopefully it's not too much. I have a backup folder of the images locally if we decide I compressed too much since I left a little wiggle room size-wise.
- Closes #342.

cthunes commented:
Interestingly, it seems installed package size differs by OS here.
My local puts me at (I think) 4.7 Mb total/3.8 Mb for /doc with no notes appearing in output
GitHub actions MacOS: 5.4Mb total/4.1Mb for /doc
GA Ubuntu: 5.4Mb total/4.0Mb for /doc
GA Windows: 5.1Mb total/3.8Mb for /doc

I guess more compression is needed? Or removing images entirely.

madison-feshuk commented:
The resultant PDF file is now much larger when plotting on this test branch compared to using dev. File is too large to share over email. 

In my example, I'm only plotting two endpoints:

> tcplPlot(type="mc", fld = "aeid", val = c(3223,3226), verbose = TRUE, output = "pdf",  fileprefix = "plots_CCTE_Deisenroth_DEVTOX_25MAR2025")

![image](https://github.com/user-attachments/assets/b124f870-5ccc-4885-8ed5-b0356136d520)

cthunes commented:
- Removed a couple images, downsized a few more as I could.
- Added links to view full res image from GitHub wiki (but not to wiki, to actual file location)
- Added links to view plot images I removed for space's sake

Final local installed size is 3.99Mb, which should put Github actions MacOS around 4.7Mb, Linux 4.7Mb, Windows 4.4Mb.

Vignette updates generally are also ready for review now @madison-feshuk

jabedini-2404 reviewed (approved):
Knit the markdown and it looks great
